flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/CreatingExecutionGraphTest.java

```diff
@@ -155,6 +179,60 @@ public void testSuccessfulSlotAssignmentTransitionsToExecuting() throws Exceptio
         }
     }
 
+    @Test
+    public void testOperatorCoordinatorUsesFailureHandlerOfTheCurrentState() throws Exception {
+        try (MockCreatingExecutionGraphContext context = new MockCreatingExecutionGraphContext()) {
+            final CompletableFuture<CreatingExecutionGraph.ExecutionGraphWithVertexParallelism>
+                    executionGraphWithVertexParallelismFuture = new CompletableFuture<>();
+            final AtomicReference<GlobalFailureHandler> operatorCoordinatorGlobalFailureHandlerRef =
+                    new AtomicReference<>();
+            new CreatingExecutionGraph(
+                    context,
+                    executionGraphWithVertexParallelismFuture,
+                    log,
+                    (executionGraph, errorHandler) -> {
+                        operatorCoordinatorGlobalFailureHandlerRef.set(errorHandler);
+                        return new TestingOperatorCoordinatorHandler();
+                    });
+
+            final StateTrackingMockExecutionGraph executionGraph =
+                    new StateTrackingMockExecutionGraph();
+
+            context.setTryToAssignSlotsFunction(CreatingExecutionGraphTest::successfulAssignment);
+            context.setExpectedExecuting(
+                    ignored -> {
+                        // This just lets us transition to executing.
+                    });
+
+            final AtomicBoolean contextGlobalFailureHandlerCalled = new AtomicBoolean(false);
+            context.setGlobalFailureHandler(
+                    t -> {
+                        contextGlobalFailureHandlerCalled.set(true);
+                    });
+
+            executionGraphWithVertexParallelismFuture.complete(
+                    CreatingExecutionGraph.ExecutionGraphWithVertexParallelism.create(
+                            executionGraph, new TestingVertexParallelism()));
+
+            operatorCoordinatorGlobalFailureHandlerRef
+                    .get()
+                    .handleGlobalFailure(new RuntimeException("Test."));
+            assertTrue(contextGlobalFailureHandlerCalled.get());
```

Comment:
This is what I meant in https://github.com/apache/flink/pull/18761#discussion_r806234796.

```suggestion
            executionGraphWithVertexParallelismFuture.complete(
                    CreatingExecutionGraph.ExecutionGraphWithVertexParallelism.create(
                            executionGraph, new TestingVertexParallelism()));

            assertThat(operatorCoordinatorGlobalFailureHandlerRef).isSameAs(context);
```

Reply:
👍 will change that